### PR TITLE
Feature/persistent replication to remote

### PR DESCRIPTION
**Pull request by jofomah**

### CODE REVIEW

File: src/components/core/core.service.js [hunk 1]
```diff
@@ -111,22 +118,78 @@ angular.module('core')
...
+      options.to.opts.query_params = {
+        docTypes: JSON.stringify(docTypes)
+      };
+
+      var replication = syncService.replicateToRemote(config.localDB, config.db, options);
+
+      replication.on('connect', function(){});
+
+      replication.on('disconnect', function (err) {
+       log.error('remoteReplicationDisconnected', err);
```
Comment (tlvince) [inline src/components/core/core.service.js]: I'm getting this error after remote sync.

Does disconnect happen after remote replication has finished?

Comment (jofomah) [inline src/components/core/core.service.js]: no it does not happen

**Comment by tlvince** [comment]
Experienced:

```
(node) warning: possible EventEmitter memory leak detected. 11 listeners added. Use emitter.setMaxListeners() to increase limit.pouchdb.js:10419 39.EventEmitter.addListenerpouchdb.js:8376 31.Changes.addListenerpouchdb.js:4878 11.api._changespouchdb.js:5357 12.Changes.doChangespouchdb.js:5270 Changespouchdb.js:696 1.AbstractPouchDB.changesangular-pouchdb.js:38 angular.module.constant.service.eventEmitterpouchdb-persist.js:80 (anonymous function)angular.js:13137 processQueueangular.js:13153 (anonymous function)angular.js:14353 $get.Scope.$evalangular.js:14169 $get.Scope.$digestangular.js:14391 (anonymous function)angular.js:4889 completeOutstandingRequestangular.js:5269 (anonymous function)
pouchdb.js:10425 console.trace()pouchdb.js:10425 39.EventEmitter.addListenerpouchdb.js:8376 31.Changes.addListenerpouchdb.js:4878 11.api._changespouchdb.js:5357 12.Changes.doChangespouchdb.js:5270 Changespouchdb.js:696 1.AbstractPouchDB.changesangular-pouchdb.js:38 angular.module.constant.service.eventEmitterpouchdb-persist.js:80 (anonymous function)angular.js:13137 processQueueangular.js:13153 (anonymous function)angular.js:14353 $get.Scope.$evalangular.js:14169 $get.Scope.$digestangular.js:14391 (anonymous function)angular.js:4889 completeOutstandingRequestangular.js:5269 (anonymous function)
```

**Comment by tlvince** [comment]
This is mostly working, besides some known bugs (item:1169, item:1170). Since we need to demo tomorrow, I'll merge this in, bearing in mind those bugs should be prioritised.

**Pull request merged**